frontend/public/components/utils/dropdown.jsx

```diff
@@ -598,6 +532,43 @@ export class Dropdown extends DropdownMixin {
   }
 }
 
+export const Dropdown = (props) => {
+  const { storageKey } = props;
+
+  // Should be undefined so that we don't save undefined-bookmarks.
+  const bookmarkStorageKey = storageKey ? `${storageKey}-bookmarks` : undefined;
+
+  const [favoriteKey, setFavoriteKey] = useUserSettingsCompatibility(
+    storageKey,
+    storageKey,
```

Comment:
The format of the user-settings keys should be `namespace.key`. eg `console.foobar`

I'd prefer to see a temporary prop in place for the old key and new one that adheres to the new format.
The current keys I see created are:
```
  dropdown-storage-namespaces
  dropdown-storage-namespaces-bookmarks
```

Reply:
Okay, this (old) Dropdown component has now a `userSettingsPrefix` and `storageKey` prop, so that the users can set both attributes for the `useUserSettingsCompatibility`. I used the prefix name because the component automatically appends `.favorite` and `.bookmarks`.